libcodechecker/libclient/credential_manager.py

```diff
@@ -71,15 +80,42 @@ def get_token(self, host, port):
         return self.__tokens.get("{0}:{1}".format(host, port))
 
     def get_auth_string(self, host, port):
-        ret = self.__save['credentials'].get('{0}:{1}'.format(host, port))
-        if not ret:
-            ret = self.__save['credentials'].get(host)
-        if not ret:
-            ret = self.__save['credentials'].get('*:{0}'.format(port))
-        if not ret:
-            ret = self.__save['credentials'].get('*')
-
-        return ret
+        """
+        hostname without protocol and product url
+        """
+        p = re.compile(
```

Comment:
Cannot we use `split_product_url` util function to get the parts of the URL?
https://github.com/Ericsson/codechecker/blob/9da592200604a1a4d6d9417ee0334e075904b341/libcodechecker/util.py#L349

Reply:
`split_product_url` needs to get a valid product url, in this case it is possible it will not be a valid product url.

Reply:
There's a `split_server_url` method too. If the URL given is not valid, I don't think it could be used for a login anyways, so we can just swallow the exception that comes out of these methods.